### ISSUE TITLE
Hooks question

### ISSUE THREAD

**Issue by Hube2** [bordoni/fakerpress#122]
I have some questions about any hooks that I can tie into when generating content on a site I'm working on. Didn't see any type of documentation on this and rather than dig through your code to see if there are hooks that will let me do what I want I figured I just ask first.

Basically, what I need to do is simulate what will happen when the client adds a specific type of post.

Before fakerpress starts adding content I need to trigger a specific hook in another plugin.
```do_action('other_plugins_hook_name');```
After fakerpress adds each post I need to trigger another hook in this plugin that includes the post ID of the post just added
```do_action('other_plugin_second_hook', $post_id);```
and finally, when fakerpress has completed adding all of the fake posts I need to trigger a final action in the other plugin.
```do_action('other_plugin_hook_3');```
So, my question is, are there any filters or actions fired in fakerpress at these three points that I can use to add filters for so that I can trigger these other hooks?

My only other alternative is to manually create enough data so that I can continue working on this custom client plugin, and like you I really hate creating test data manually.

**Comment by bordoni** [comment]
Hi @Hube2,

Sorry for the delay on my response, currently I only have Hooks for the First and Second request.

The actual saving happens on here on this Filter, on priority `10`:
`fakerpress.module.{$module_name}.save`

For the before starting to add any content you can use, on a lower priority than 10:
`wp_ajax_fakerpress.module_generate`

Here is the catch, I am going to refactor a bunch of the Hooks in the plugin to allow a lot more flexibility by third party developers. Most likely on version `0.4.11` or `0.5.0`

**Comment by Hube2** [comment]
No worries about the late reply, I actually forgot about this myself. I believe that at the time I decided I only needed a few post to work with. The code was needed when I wanted to simulate what happened during a product import. Thanks for the rely.
